console/backend/src/main/java/nl/nn/adapterframework/web/filters/CorsFilter.java

```diff
@@ -0,0 +1,152 @@
+/*
+   Copyright 2022-2023 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.web.filters;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+
+import nl.nn.adapterframework.util.StringUtil;
+
+public class CorsFilter implements Filter {
+	private final Logger secLog = LogManager.getLogger("SEC");
+	private final Logger log = LogManager.getLogger(this);
+	private String hostAddress;
+
+	@Value("${iaf-api.cors.allowOrigin:'*'}")
+	private String allowedCorsOrigins; //Defaults to nothing
+
+	@Value("${iaf-api.cors.exposeHeaders:Allow, ETag, Content-Disposition}")
+	private String exposedCorsHeaders;
+
+	//TODO: Maybe filter out the methods that are not present on the resource? Till then allow all methods
+	@Value("${iaf-api.cors.allowMethods:GET, POST, PUT, DELETE, OPTIONS, HEAD}")
+	private String allowedCorsMethods;
+
+	@Value("${iaf-api.cors.enforced:false}")
+	private boolean enforceCORS;
+
+	private List<String> allowedCorsDomains =  new ArrayList<>();
+
+	@Override
+	public void init(FilterConfig filterConfig) throws ServletException {
+		try {
+			hostAddress = InetAddress.getLocalHost().getHostAddress();
+		} catch (UnknownHostException e) {
+			log.warn("unable to determine local host address", e);
+		}
+
+		List<String> allowedOrigins = StringUtil.split(allowedCorsOrigins);
+		for (String domain : allowedOrigins) {
+			if(domain.startsWith("http://")) {
+				log.warn("cross side resource domain ["+domain+"] is insecure, it is strongly encouraged to use a secure protocol (HTTPS)");
+			}
+			if(!domain.startsWith("http://") && !domain.startsWith("https://")) {
```

Comment:
Wouldn't this kind of break it with a `allowedCorsOrigins` value of `*` which looks to be the default?